### ISSUE TITLE
Adding `engines` to `package.json` was a BC break

### ISSUE THREAD

**Issue by ooflorent** [final-form/final-form#434]
### Are you submitting a **bug report** or a **feature request**?

Bug report

### What is the current behavior?

Doing a minor update of `final-form` breaks the build when using node >= 16 because of:
```json
// package.json
// …
  "engines": {
    "node": "14"
  },
```

### What is the expected behavior?

Doing a minor update of `final-form` should not break the build.

### Sandbox Link

Not relevant

### What's your environment?

```
% node -v
v16.9.1
```

### Other information

```
bocms master % yarn upgrade-interactive --latest
yarn upgrade-interactive v1.22.11
info Color legend :
 "<red>"    : Major Update backward-incompatible updates
 "<yellow>" : Minor Update backward-compatible features
 "<green>"  : Patch Update backward-compatible bug fixes
? Choose which packages to update.
? Choose which packages to update. final-form@4.20.3
info Installing "dependencies"...
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
error final-form@4.20.3: The engine "node" is incompatible with this module. Expected version "14". Got "16.9.1"
✨  Done in 59.64s.
```

**Comment by zburke** [comment]
This also breaks for folks on node < 14. Node v12 is still in maintenance mode through mid 2022.
> error final-form@4.20.3: The engine "node" is incompatible with this module. Expected version "14". Got "12.22.2"
> error Found incompatible module.

**Comment by erikras** [comment]
Oh no!! I'll look in asap.

**Comment by erikras** [comment]
Should be fixed in [`v4.20.4`](https://github.com/final-form/final-form/releases/tag/v4.20.4). Can you confirm?

**Comment by zburke** [comment]
@erikras, Yay, `v4.20.4` looks good! ... But I'm now hung up by `react-final-form` `v6.5.4`. I'll file a bug over there too :) Thanks for the speedy attention!
